tests/CodeControllerTest.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EWS;
+
+[TestClass]
+public class CodeControllerTest
+{
+    private EwsContext context;
+    private CodeController controller;
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        context = ContextFactory.CreateContext();
+        controller = new CodeController(context);
+    }
+
+    [TestCleanup]
+    public void TestCleanup()
+    {
+        context.Dispose();
+    }
+
+    [TestMethod]
+    public async Task GetByCodeTypAsync()
+    {
+        var codeId = 303;
+
+        // get codes with codetype 1
+        var codes = await controller.GetByCodeTypAsync(1).ConfigureAwait(false);
```

Comment:
PP: Alle Codetypen in ein Enum auslagern und die Werte entsprechend der DB vergeben. Die Werte sind stabil und werden nie ändern und mit einem Enum macht es vieles nachvollziehbarer.

Reply:
Vielleicht wäre es eine Überlegung wert die Codes und Codetypen etc. (=> alles was mit der App nicht editiert werden kann und stabil bleibt) in unseren Testdaten bereits entsprechend den Originaldaten anzulegen. So werden einerseits die Dropdowns in den Masken beim schon beim Entwickeln logischer und andererseits können wir die Enums gleich richtig benennen.

Reply:
Ja, das wäre sinnvoll.